### ISSUE TITLE
[BUG] SOLR_RELATION alwas in default language

### ISSUE THREAD

**Issue by npoggenburg** [TYPO3-Solr/ext-solr#2504]
**Describe the bug**
Queue fields from type SOLR_RELATION are alwas in the language you picked in the TYPO3 Backend.

**To Reproduce**
1. Install news
2. add any kind of mm relation, for example a product record
3. Add record in backend "Product(Lang:EN)" and translate it to "Product(Lang:DE)"
4. Select the record "Product(Lang:EN)" in your news "News(Lang:EN)"
5. Translate your news "News(Lang:EN)" to "News(Lang:DE)"
6. The translated news "News(Lang:DE)" got the record "Product(Lang:EN)" selected in backend. because this is how TYPO3 works.
7. Index the News records with SOLR.

**Expected behavior**
When you add the product record to your news queue:
```
product_stringM = SOLR_RELATION
product_stringM {
    localField = product_items
    multiValue = 1
}
```
You would expect that the news "News(Lang:EN)" field product_stringM would have the value of "Product(Lang:EN)" and the "News(Lang:DE)" would have the value of "Product(Lang:DE)", but both got "Product(Lang:EN)".

**Screenshots**
![Bildschirmfoto 2019-11-18 um 14 42 43](https://user-images.githubusercontent.com/18659807/69057290-e15e5900-0a11-11ea-92e3-1bb0c6a13cff.png)
![Bildschirmfoto 2019-11-18 um 14 42 09](https://user-images.githubusercontent.com/18659807/69057291-e1f6ef80-0a11-11ea-8bd1-bac54f79b87f.png)
![Bildschirmfoto 2019-11-18 um 14 39 05](https://user-images.githubusercontent.com/18659807/69057293-e1f6ef80-0a11-11ea-8eed-3415e9aa001b.png)
![Bildschirmfoto 2019-11-18 um 14 37 50](https://user-images.githubusercontent.com/18659807/69057294-e1f6ef80-0a11-11ea-838c-ddfa7dbc0f8e.png)

**Used versions:**
 - TYPO3 Version: 9.5.11
 - EXT:solr Version: 9.0.2

Is there any working example on how to use SOLR with a TYPO3 multilanguage site in STRICT mode?

**Comment by majernik** [comment]
The problem exists under TYPO3 8.7.27 and Solr 8.1.2 as well. The sys_language_uid is not always set to the language, that is being currently indexed, when initializing TSFE via initializeTsfe method in Utils class.

If I set
`$GLOBALS['TSFE']->sys_language_uid = $language; `
at the end, then the relation is indexed correctly.
Note: This is not a solution as no language fallback is applied here.

**Comment by JesperBom** [comment]
I can confirm. Same issue here.

We got an english site, with a french translation (strict mode)
When we index english using a SOLR_RELATION it is all fine.
But when indexing french it won't index the french relations but english instead.
We have an mm relation on the page:
`'tx_sitetemplate_highlights' => [
            'label' => $translation . 'column.highlights',
            'config' => [
                'allowed' => Bullet::TABLE,
                'appearance' => [
                    'enabledControls' => [
                        'delete' => true,
                        'dragdrop' => true,
                        'hide' => true,
                        'info' => false,
                        'localize' => false,
                        'new' => true,
                        'sort' => true,
                    ],
                    'newRecordLinkAddTitle' => false,
                    'newRecordLinkTitle' => $translation . 'column.highlights.add',
                    'showAllLocalizationLink' => false,
                    'showPossibleLocalizationRecords' => false,
                    'showRemovedLocalizationRecords' => false,
                    'showSynchronizationLink' => false,
                    'useSortable' => true,
                ],
                'behaviour' => [
                    'allowLanguageSynchronization' => true,
                ],
                'type' => 'inline',
                'foreign_table' => Bullet::TABLE,
                'MM' => Bullet::MM_TABLE,
                'MM_opposite_field' => 'usage_mm',
                'MM_match_fields' => [
                    'fieldname' => 'tx_sitetemplate_highlights',
                    'tablename' => $table,
                ],
            ],
        ],`

This does work in typo3 itself, english works, french works.

But when indexing the page with solr it always gets the english values instead of the french ones:
solr field config:
`            highlights_stringM = SOLR_RELATION
            highlights_stringM {
                localField = tx_sitetemplate_highlights
                multiValue = 1
            }`

I tried to change to fallback language mode, this caused alot of errors, this mode is also not wanted project wise

**Comment by dkd-kaehm** [comment]
We can fix that issue on EXT:solr 11.5.x only, because of complexity and deep interlacement of TSFE usages and initialization in previous versions.

### Fixed in !!![TASK] Improve and Fix TSFE Initialization https://github.com/TYPO3-Solr/ext-solr/commit/a246cb8e34d80da6227445c0c800620f16264a00
